alvr/client_core/src/connection.rs

```diff
@@ -468,32 +465,22 @@ async fn stream_pipeline(
                     break Ok(());
                 }
 
-                let header = VideoFrame {
-                    packetCounter: packet.header.packet_counter,
-                    trackingFrameIndex: packet.header.tracking_frame_index,
-                    videoFrameIndex: packet.header.video_frame_index,
-                    sentTime: packet.header.sent_time,
-                    frameByteSize: packet.header.frame_byte_size,
-                    fecIndex: packet.header.fec_index,
-                    fecPercentage: packet.header.fec_percentage,
-                };
-
                 if let Some(stats) = &mut *STATISTICS_MANAGER.lock() {
                     stats.report_video_packet_received(Duration::from_nanos(
                         packet.header.tracking_frame_index,
                     ));
                 }
 
-                let mut fec_failure = false;
-                unsafe {
-                    crate::processNalPacket(
-                        header,
-                        packet.buffer.as_ptr(),
-                        packet.buffer.len() as _,
-                        &mut fec_failure,
-                    )
-                };
-                if fec_failure {
+                video_frame_buffer.set_frame_lost(packet.had_packet_loss);
+                video_frame_buffer.push(packet.header, packet.buffer);
```

Comment:
Ultimately, I'd request decoder buffer for each new frame and copy the packets directly from socket.

Reply:
maybe add a note of that there? to explain why in case of someone in the future seeing this and wondering,